docs/rules/test-title-format.md

```diff
@@ -0,0 +1,67 @@
+# Ensure tests have a correct title format
+
+Tests should have a title matching the format option.
```

Comment:
I think we can just drop this line. It doesn't add much over the title. Instead, would be good if you could mention some use-cases for the rule.